client/state/reader/site-blocks/actions.js

```diff
@@ -24,11 +30,16 @@ export function requestSiteBlock( siteId ) {
 			siteId
 		} );
 
-		return wpcom.undocumented().me().blockSite( siteId ).then( ( data ) => {
+		return wpcom.undocumented().me().blockSite( siteId ).then( ( response ) => {
+			if ( response && response.success === false ) {
+				return Promise.reject( 'Block was unsuccessful' );
+			}
+			return response;
+		} ).then( ( response ) => {
```

Comment:
on second thought, dont do that.
`payload` isn't compatible with `keyedReducer` (yet).